plugin/storage/cassandra/spanstore/dbmodel/converter.go

```diff
@@ -83,6 +92,10 @@ func (c converter) toDomain(dbSpan *Span) (*model.Span, error) {
 	if err != nil {
 		return nil, err
 	}
+	warnings, err := c.fromDBWarnings(dbSpan.Tags)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
this is not a check, it's error handing. If `c.fromDBWarnings` return an error we have to react accordingly. And then we should test that we indeed react accordingly. You can see other tests that cause fromDBTags to fail and simulate this error condition.

Reply:
right, what I meant was there's only one point where this method (`c.fromDBWarnings`) can return an error, i.e. when the one of the tags passed as an argument has an invalid `ValueType`. Since we're calling `c.fromDBWarnings` _after_ `c.fromDBTags`, and both the methods accept the same arguments, handle the same errors - the above lines may never be executed.

Reply:
you can change the order in those methods to first test the key for a prefix, and only then try to convert the tag from db

Reply:
this way the two methods will react to different tags in the list

Reply:
Got it. I've updated the changes accordingly.